consensus/core/src/authority_node.rs

```diff
@@ -180,6 +185,22 @@ impl NetworkService for AuthorityService {
             return Err(e);
         }
         let verified_block = VerifiedBlock::new_verified(signed_block, serialized_block);
+
+        // Reject block with timestamp too far in the future.
+        let forward_time_drift =
+            Duration::from_millis(timestamp_utc_ms().saturating_sub(verified_block.timestamp_ms()));
```

Comment:
I think the operation should be instead `Duration::from_millis(verified_block.timestamp_ms().saturating_sub(timestamp_utc_ms());`  in order to detect if the block timestamp is indeed ahead in future.

Reply:
Ah definitely.